src/components/repeat/repeat.test.tsx

```diff
@@ -0,0 +1,152 @@
+
+    expect(updatedExpression).toEqual(
+      "myValue > today() || myValue <= '1/1/1890' || myValue > useFieldValue('visit_date') || myValue < useFieldValue('visit_date')",
+    );
+  });
+});
+
+describe('Repeat Component Tests', () => {
+  const mockUsePatient = jest.mocked(usePatient);
+  const mockUseSession = jest.mocked(useSession);
+
+  global.ResizeObserver = require('resize-observer-polyfill');
+
+  jest.mock('@openmrs/esm-framework', () => {
+    const originalModule = jest.requireActual('@openmrs/esm-framework');
+    return {
+      ...originalModule,
+      usePatient: jest.fn(),
+      useSession: jest.fn(),
+      createGlobalStore: jest.fn(),
+      ActionMenu: jest.fn(() => <div />),
+    };
+  });
+
+  jest.mock('../../provider/form-provider', () => {
+    const originalModule = jest.requireActual('../../provider/form-provider');
+    return {
+      ...originalModule,
+      useFormProviderContext: jest.fn(),
+    };
+  });
+
+  jest.mock('../../api', () => ({}));
+
+  const renderForm = async (mode: SessionMode = 'enter') => {
+    await act(async () => {
+      render(
+        <FormEngine
+          formJson={repeatingComponentTestForm as FormSchema}
+          patientUUID="8673ee4f-e2ab-4077-ba55-4980f408773e"
+          mode={mode}
+          encounterUUID={mode === 'edit' ? 'a8817ad2-ef92-46c8-bbf7-db336505027c' : null}
+        />,
+      );
+    });
+  };
+
+  const user = userEvent.setup();
+
+  const mockContext = {
+    patient: {},
+    sessionMode: 'enter',
+    formFields: repeatingComponentTestForm.pages[0].sections[0].questions,
+    methods: { getValues: jest.fn(), setValue: jest.fn() },
+    addFormField: jest.fn(),
+    formFieldAdapters: { obsGroup: { transformFieldValue: jest.fn() } },
+  };
+
+  beforeEach(() => {
+    Object.defineProperty(window, 'i18next', {
+      writable: true,
+      configurable: true,
+      value: {
+        language: 'en',
+        t: jest.fn(),
+      },
+    });
+
+    mockUsePatient.mockImplementation(() => ({
+      patient: mockPatient,
+      isLoading: false,
+      error: undefined,
+      patientUuid: mockPatient.id,
+    }));
+
+    mockUseSession.mockImplementation(() => mockSessionDataResponse.data);
+    (useFormProviderContext as jest.Mock).mockReturnValue(mockContext);
+  });
+
+  it('Should add a repeatable field instance on clicking "Add"', async () => {
+    await renderForm();
+    const addButton = screen.getByText(/add/i);
+    await user.click(addButton);
+
+    expect(mockContext.addFormField).toHaveBeenCalledTimes(1);
+    expect(mockContext.addFormField).toHaveBeenCalledWith(expect.objectContaining({ id: 'patientContact_1' }));
+  });
+
+  it('Should clone the field at origin on clicking "Add"', async () => {
+    await renderForm();
+    const addButton = screen.getByText(/add/i);
+    await user.click(addButton);
+
+    const clonedField = screen.getByLabelText(/Contact relationship/i);
+    expect(clonedField).toBeInTheDocument();
```

Comment:
Can you confirm that there are two fields with the label "Contact relationship" that never existed on the initial render?